src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs

```diff
@@ -796,30 +792,37 @@ private void TagIcon_Tapped(object sender, TappedRoutedEventArgs e)
 			e.Handled = true;
 		}
 
-		private void ItemRow_PointerEntered(object sender, PointerRoutedEventArgs e)
+		private void IconBox_PointerEntered(object sender, PointerRoutedEventArgs e)
+		{
+			UpdateCheckboxVisibility((sender as FrameworkElement)!.FindAscendant<ListViewItem>()!, true);
+		}
+
+		private void IconBox_PointerExited(object sender, PointerRoutedEventArgs e)
 		{
-			UpdateCheckboxVisibility(sender, true);
+			e.Handled = true;
+		}
+
+		private void IconBox_PointerCanceled(object sender, PointerRoutedEventArgs e)
+		{
+			e.Handled = true;
 		}
```

Comment:
I don't think we really need them

Reply:
It is necessary. Without it, the row highlight will momentarily disappear when the checkbox appears.